src/engine/api.rs

```diff
@@ -17,6 +17,129 @@ pub fn next_connection_id() -> ConnectionId {
 }
 
 #[derive(Debug, PartialEq, Clone)]
+pub enum ClientMessage {
+    Consumer(ConsumerManagerMessage),
+    Producer(ProducerManagerMessage),
+    Both(ConsumerManagerMessage, ProducerManagerMessage),
+}
+unsafe impl Send for ClientMessage {}
```

Comment:
huh.... I had no idea that Send and Sync were automatically implemented. That's really convenient! Honestly, I though that having to put `unsafe impl Send` everywhere was kind of a wart on the language. Glad to know it's just my own shoddy understanding ;)